dlrs/main/classes/RollupSummariesSelector.cls

```diff
@@ -363,37 +426,164 @@ public class RollupSummariesSelector {
     public List<RollupSummary> selectActiveByUniqueName(
       Set<String> uniqueNames
     ) {
-      return RollupSummary.toList(
-        Database.query(
-          newQueryFactory()
-            .setCondition('DeveloperName in :uniqueNames and Active__c = true')
-            .addOrdering(
-              LookupRollupSummary2__mdt.ParentObject__c,
-              fflib_QueryFactory.SortOrder.ASCENDING
-            )
-            .addOrdering(
-              LookupRollupSummary2__mdt.RelationshipField__c,
-              fflib_QueryFactory.SortOrder.ASCENDING
-            )
-            .toSOQL()
-        )
-      );
+      if (this.forEdit) {
+        return RollupSummary.toList(
+          this.getLookupRollupSummaries(
+            newQueryFactory()
+              .setCondition('DeveloperName in :tmpVar1 and Active__c = true')
+              .addOrdering(
+                LookupRollupSummary2__mdt.ParentObject__c,
+                fflib_QueryFactory.SortOrder.ASCENDING
+              )
+              .addOrdering(
+                LookupRollupSummary2__mdt.RelationshipField__c,
+                fflib_QueryFactory.SortOrder.ASCENDING
+              )
+              .toSOQL(),
+            new List<Object>{ uniqueNames }
+          )
+        );
+      } else {
+        List<RollupSummary> summaries = new List<RollupSummary>();
+        List<SObject> records = new List<SObject>();
+        for (String uniqueName : uniqueNames) {
+          SObject record = getLookupRollupSummaryByDeveloperNameOrId(
+            uniqueName
+          );
+          if (record != null && (Boolean) record.get('Active__c')) {
+            records.add(record);
+          }
+        }
+
+        return sortSummaries(
+          RollupSummary.toList(records),
+          'ParentObject__c',
+          'RelationshipField__c'
+        );
+      }
     }
 
     /**
      * Select all
      **/
     public List<RollupSummary> selectAll() {
-      return RollupSummary.toList(
-        Database.query(
-          newQueryFactory()
-            .addOrdering(
-              LookupRollupSummary2__mdt.Label,
-              fflib_QueryFactory.SortOrder.ASCENDING
-            )
-            .toSOQL()
-        )
-      );
+      if (this.forEdit) {
+        return RollupSummary.toList(
+          this.getLookupRollupSummaries(
+            newQueryFactory()
+              .addOrdering(
+                LookupRollupSummary2__mdt.Label,
+                fflib_QueryFactory.SortOrder.ASCENDING
+              )
+              .toSOQL(),
+            new List<Object>()
+          )
+        );
+      } else {
+        List<RollupSummary> summaries = RollupSummary.toList(
+          this.getLookupRollupSummaries()
+        );
+
+        return sortSummaries(summaries, 'Label', null);
+      }
+    }
+
+    protected virtual List<SObject> getLookupRollupSummaries(
```

Comment:
This method is overloaded a few times, but reading the code where it's used it may be better to have separate names for these. `executeSoqlWithParams` for this one perhaps?